platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,826 @@
+* The complete implementation of the TSS-Library is required for the TSS-Ledger-ID proposal to be fully implemented.
+* Consensus nodes must have their long-term EC keys generated and stored in the address book in a release prior to the
+  TSS-Ledger-ID proposal being delivered.
+
+Dependencies on the TSS-Roster
+
+* At minimum, the `Roster` API is defined with an AddressBook wrapper for initial development and testing.
+* Delivery of TSS-Ledger-ID will require that the TSS-Roster proposal is delivered in a prior release.
+
+Impacts of TSS-Ledger-ID to Services Team
+
+* The timing of submitting the candidate roster may need to be adjusted to allow enough time for the candidate
+  roster to have key material generated.
+* Services will need to know or detect when the candidate roster fails to be adopted due to not having enough key
+  material generated.
+* The node's long term EC public key will need to be added to the address book.
+    * This requires modification of the HAPI transactions that manage the address book.
+
+Impacts to DevOps Team
+
+* Each consensus node will need a new long term EC key in addition to the existing RSA key.
+    * EC Key generation will have to happen before a node can join the network.
+* A new node added to an existing network will need to be given a state from an existing node after the network has
+  adopted the address book containing the new node.
+
+Implications Of Completion
+
+* The consensus nodes will be able to create ledger signatures for use in block proofs.
+* TSS-Block-Signing proposal becomes unblocked for development.
+
+### Requirements
+
+TSS Core Requirements
+
+| Requirement ID | Requirement Description                                                                                                      |
+|----------------|------------------------------------------------------------------------------------------------------------------------------|
+| TSS-001        | The ledger MUST have a public/private signing key pair that does not change when the consensus roster changes.               |
+| TSS-002        | The ledger private key MUST be a secret that nobody knows.                                                                   |
+| TSS-003        | The `ledger id` MUST be a public key that everyone knows and able to verify signatures from the ledger private key.          |
+| TSS-004        | The ledger signature for a message MUST be produced through a threshold number of consensus nodes signing the message.       |
+| TSS-005        | There MUST be a way to bootstrap an existing network's consensus roster with to give the ledger a public/private signing key |
+| TSS-006        | There MUST be a way for a new network to generate a public/private signing key for the ledger.                               |
+| TSS-007        | There MUST be a way to rotate the ledger's public/private signing key.                                                       |
```

Comment:
No one has talked about it significantly, but I'm putting it in there because without it there is no way to recover from the situation where a anyone, let alone a bad actor, gets ahold of the ledger private key.    Services would need to create a new HAPI transaction that would require GC sign-off that would trigger this event:  break-glass-in-case-of-emergency.   I think someone getting ahold of the ledger private key is worse than a catastrophic ISS.    They could sign whatever they want with it.

Reply:
I'm going to leave the services side of it to the tss-block-signing proposal.

Reply:
>  I think someone getting ahold of the ledger private key is worse than a catastrophic ISS

I would agree. IMO this isn't something that would be recoverable, it would be the end of the network (and thus doesn't warrant consideration). Though other's could disagree. I think leaving it to services is a reasonable choice.